src/common/directives/imageUpload.directive.js

```diff
@@ -0,0 +1,64 @@
+import angular from 'angular'
+
+const directiveName = 'imageUpload'
+
+// Based on https://github.com/leon/angular-upload
+// Fixes bug where even if accept is "image/*", you could still drag-and-drop non-image files onto the upload button
+const imageUpload = /* @ngInject */ ($http) => ({
+  restrict: 'EA',
+  scope: {
+    url: '@',
+    onUpload: '&',
+    onSuccess: '&',
+    onError: '&',
+    onComplete: '&'
+  },
+  link: (scope, element) => {
+    const validMimeTypes = ['image/jpeg', 'image/png']
+
+    const el = angular.element(element)
+    const fileInput = angular.element(`<input type="file" accept="${validMimeTypes.join(',')}" />`)
+    el.append(fileInput)
+
+    fileInput.on('change', (event) => {
+      const files = event.target.files
+      const file = files[0]
+      if (!file) {
+        return
+      }
+
+      if (!validMimeTypes.includes(file.type)) {
```

Comment:
We should probably write multiple tests for this, 1 for each supported file type.